src/isolation/tempdir.rs

```diff
@@ -1,25 +1,132 @@
-use std::{fs::Permissions, os::unix::fs::PermissionsExt, path::PathBuf};
+use std::{
+	fs::{Permissions, create_dir, set_permissions},
+	os::unix::fs::{MetadataExt, PermissionsExt},
+	path::PathBuf,
+};
 
 use tempfile::{Builder, TempDir, env};
 use uuid::Uuid;
 
+/// Gets own user ID to append to subdirectory (e.g. `/tmp/uhyve-2`).
+///
+/// Defaults to 0 if accessing own user ID is not possible and hopes
+/// for the best.
+fn get_uid() -> u32 {
+	// Use user ID to create subdirectory.
+	std::fs::metadata("/proc/self")
```

Comment:
Wouldn't something like https://docs.rs/libc/latest/libc/fn.getuid.html be more appropriate?

Reply:
I wanted to avoid the unsafe call, yet `getuid` is never supposed to fail. When thinking about it more deeply, this might be the better way to go.